src/olympia/hero/admin.py

```diff
@@ -58,6 +66,21 @@ class Media:
     actions = ['delete_selected']
     readonly_fields = ('preview_image',)
 
+    def save_model(self, request, obj, form, change):
+        super().save_model(request, obj, form, change)
+        (path, fn) = os.path.split(obj.custom_image.path)
+        dest_thumb = path + b'/thumbs/' + fn
+
+        size_thumb = (150, 120)
+        size_full = (960, 640)
+
+        img = Image.open(obj.custom_image)
+        f = tempfile.NamedTemporaryFile(dir=settings.TMP_PATH)
+        img.save(f, 'png')
+
+        resize_image(f.name, dest_thumb, size_thumb)
+        resize_image(f.name, obj.custom_image.path, size_full)
```

Comment:
How about:
```suggestion
        resize_image(obj.custom_image.path, dest_thumb, size_thumb)
        with tempfile.NamedTemporaryFile(dir=settings.TMP_PATH) as tmp:
            resize_image(obj.custom_image.path, tmp.name, size_full)
            copy_stored_file(tmp.name, obj.custom_image.path)
```
(untested!)

Reply:
Ah I think I see now, so I should've written `resize_image(obj.custom_image.path, f.name, size_full)` instead and then moved it with `copy_stored_file(f.name, obj.custom_image.path)`, correct?

Reply:
yeah.  Using `with` means that the temporary file is closed (and deleted) at the end of the `with` block so we don't need to worry about remembering to close it (and handle exceptions leaving the file open, etc).